### DISCUSSION TITLE
What improvements have been made to SSL in cefsharp98.1.190? Some websites cannot be opened

### DISCUSSION THREAD

**Discussion by cdz1980** [cefsharp/CefSharp#4510] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

98.1.190

### Operating System

Windows 10

### Architecture

x64

### .Net Version

.net 4.8

### Implementation

WinForms

### Reproduction Steps

 var settings = new CefSettings();
            settings.IgnoreCertificateErrors = true;
            Cef.Initialize(settings);

 ChromiumWebBrowser  browser = new ChromiumWebBrowser("https://vms.whirlpool-china.com:7300/sanyoprod/mainframework/rpt/ims_repair_taskslist.jsp");
            browser.Dock = DockStyle.Fill;
            this.Controls.Add(browser);

### Expected behavior

Open website normally

### Actual behavior

The website cannot be opened, it is completely blank

### Regression?

Going back to version 97.1.6 worked and no code changes were necessary.

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

### Other information

CEFSharp version 98.1.190 and above cannot access https://vms.whirlpool-china.com:7300/sanyoprod/mainframework/rpt/ims_repair_taskslist.jsp normally, despite setting IgnoreCertificateErrors to true using settings.IgnoreCertificateErrors. However, https://wrong.host.badssl.com/ loads normally, indicating that the IgnoreCertificateErrors function is effective. Additionally, downgrading to version 97.1.6 of CEFSharp allows access to the website. What can be done to resolve this problem? Please translate and optimize the above content.

The following error occurred when using cefClient
Error: ERR_SSL_VERSION_OR_CIPHER_MISMATCH (-113)
ERR_SSL_VERSION_OR_CIPHER_MISMATCH

**Comment by aimcosoftware** [comment]
You probably need this: https://cefsharp.github.io/api/111.2.x/html/M_CefSharp_IRequestHandler_OnCertificateError.htm
It also allows you to selectively ignore.

**Comment by amaitland** [comment]
> Please only open an issue if you can reproduce the problem with version 112.3.0 or later.

As per the bug report template. Please **don't** open issues for old unsupported versions.

> What improvements have been made to SSL in cefsharp98.1.190?

`SSL/TLS` are implemented at the `Chromiuim` level. You'd have to investigate what changed in `Chromium`.

> What can be done to resolve this problem?

Try upgrading to the current stable version (`M113` at time of writing). https://github.com/cefsharp/CefSharp.MinimalExample is available for testing purposes.

[OnCertificateError](https://cefsharp.github.io/api/111.2.x/html/M_CefSharp_Handler_RequestHandler_OnCertificateError.htm) is the preffered approach for allowing certificates see https://github.com/cefsharp/CefSharp/wiki/General-Usage#ssltls-certificate-errors

**Comment by cdz1980** [comment]
After investigation, I found that the website with issues is using TLS1.0. Does CefSharp version 98.1.190 no longer support TLS1.0? If so, is there any other solution besides downgrading the CefSharp version?







                       
原始邮件
                       
                     

发件人："Alex Maitland"< ***@***.*** &gt;;

发件时间：2023/5/28 4:49

收件人："cefsharp/CefSharp"< ***@***.*** &gt;;

抄送人："cdz1980"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

主题：Re: [cefsharp/CefSharp] What improvements have been made to SSL incefsharp98.1.190? Some websites cannot be opened (Discussion #4510)




  
Please only open an issue if you can reproduce the problem with version 112.3.0 or later.
  
As per the bug report template. Please don't open issues for old unsupported versions.
  
What improvements have been made to SSL in cefsharp98.1.190?
  
SSL/TLS are implemented at the Chromiuim level. You'd have to investigate what changed in Chromium.
  
What can be done to resolve this problem?
  
Try upgrading to the current stable version (M113 at time of writing). https://github.com/cefsharp/CefSharp.MinimalExample is available for testing purposes.
 
OnCertificateError is the preffered approach for allowing certificates see https://github.com/cefsharp/CefSharp/wiki/General-Usage#ssltls-certificate-errors
 
&mdash;

**Comment by amaitland** [comment]
`TLS 1.0/1.1` support was removed from `Chromium`. 

https://chromestatus.com/feature/5759116003770368
